### DISCUSSION TITLE
Which pins may be used for the STOP-button???

### DISCUSSION THREAD

**Discussion by HJLK** [rern/rAudio#1566] [Testing] [answered]
@rern: starting from scratch with the 202401 image, and manually setting it up (backup.gz is not working in /boot) it all goes wrong with the setup of the STOP-button: as I activate pin 32 the machine does immediately shutdown! Wheb rebooting, it also shuts down directly after the startup. What's happening here? Which PINs (board numbering) may be used for the STOP-button? And what does the RESERVED-pin mean?
Thank you in advance for the answers.

**Comment by R101G** [comment]
@rern Same problem using the default settings for the power button. It boots up and then shuts down.

**Comment by rern** [comment]
Direct access to GPIO pins with sysfs is obsolete in latest kernel.
All relevant scripts have been revised to utilize new commands.
Please try the UPDATE.

**Reply by R101G** [reply]
@rern The UPDATE did not fix this, but I think there is something wrong with the default setting. It refers to pin3, rather than pin5 (gpio3), and the led pin used to be 40? I cannot change the on pin.

**Reply by HJLK** [reply]
@rern: 
1) should we UPDATE to 20240113? Or must we try the UPDATE-branch? 
2) and when I update the OFF-pin to, say, 7 I see RESERVED 29, which can also be altered. What is the meaning of this?
Thanks for your time.

**Comment by HJLK** [comment, marked as answer]
OK, I updated to 20240113, and did a regression test after rebooting. The picture of radiostations did not change, nor did the LCD display, when I switched to another line in the playlist. After I did an update to the UPDATE-branch, that was corrected.
Now let us see what the STOP-button does... STOP=pin7, LED=pin11:  LED gets ON, and reboot: now STOP-button works, START-button works! 
BUT: after startup  (with Radio Paradise Mellow Mix auto started) I see the picture of the song playing.  When switching to another item in my playlist the GUI (browser) is not refreshed. Sometimes the picture is the default rAudio-picture. NB the STATION-picture is shown when the music stops. I can't get what goes exactly wrong here, because sometimes all is OK!  The app GUI has the same problem however.
If this continues to be a problem, I will make a new bug-report. As far this bug can be closed.

**Comment by R101G** [comment]
If the start button works, presumably pin3 must refer to gpio3 (pin5), as that is a hardware function? Or am I wrong?
It might be that having the led default to gpio3 is causing the problem? What does "reserved" do?

**Reply by HJLK** [reply]
@R101G: the pin-numbers used by rAudio are all **board**-numbers (connector on the Pi-board), and _not_ GPIO-numbers. You should not use a number of pins because they are used for I2C (LCDChar) and I2S (for Hifiberry DAC). I use pin7 for the STOP-button, and pin11 for the LED. The START-button is always pin3 (hardware bound). This pin is also used for I2C (so do not use it when rAudio is UP!). When the Pi-board is stopped, it can be booted by making pin3 shortly LOW. 
NB1: the STOP-pin is probably "floating" (= very high impedance!) and therefore vulnerable for noise signals, in particular when there is a long wire to the button. rAudio may be STOPped irregularly by this "feature". To be safe, connect a resistor of, say, 47kOhm from pin7 to the 3,3V-rail.  
NB2: I also do not know what the "**reserved-pin**" means, @rern did not yet give explanation...

Greetings, hope this helps!

**Reply by R101G** [reply]
@HJLK  I am pretty sure the start pin is pin5, not pin3. Grounding pin5 will always start the Pi whatever the settings, but now if the power option is enabled without changing the defaults, it all goes wrong. It's normal to toggle pin5 as the start/stop pin (this was the old rAudio default) by adding _dtoverlay=gpio-shutdown_ to config.txt . The I2c function can usually be moved to another pin if it is needed, as any connected board will have pull-up resistors on the lines.

**Reply by HJLK** [reply]
@R101G : Oops... Of course you are right! Sorry, I was confused myself! Pin5 is the hardware bound START-pin (=GPIO3). 
Still it all works as expected when I use pin7 for STOP, and pin11 for the LED. When using the _default_ (pin5) it all goes wrong, also for me. I have a CharLCD connected via a serializer and the standard Pi I2C, so I cannot use pin5 for the STOP button.
